asm/Kptfile

```diff
@@ -173,7 +173,7 @@ openAPI:
           - marker: PROJECT_ID
             ref: '#/definitions/io.k8s.cli.setters.gcloud.core.project'
           - marker: PROJECT_NUMBER
-            ref: '#/definitions/io.k8s.cli.setters.gcloud.project.environProjectNumber'
+            ref: '#/definitions/io.k8s.cli.setters.gcloud.project.projectNumber'
```

Comment:
It doesn't look like you actually use this, do you?

Reply:
You mean within the asmcli kpt commands? gcloud.project.projectNumber is set automatically after the package is downloaded (not quite clear about the logic behind it, it is similar to gcloud.core.project).

Reply:
Let me update this PR to explicitly set this field.

Reply:
Hi @zerobfd, I retested with the current PR. According to the following log, ``` 2021-09-30T22:46:14.170951 asmcli: Running: '/usr/local/google/home/tairan/go/src/asm/asm-helper/test/gcp/fleet/tmp1/kpt cfg set asm gcloud.core.project test-multiproj1'
2021-09-30T22:46:14.239829 asmcli: -------------
asm/
set 19 field(s) of setter "gcloud.core.project" to value "test-multiproj1"
asm/
set 2 field(s) of setter "gcloud.project.projectNumber" to value "416604968218"
```, looks like when `kpt cfg set asm gcloud.core.project test-multiproj1` is executed, `kpt cfg set asm gcloud.project.projectNumber` is executed automatically. Thus I do not need to call it explicitly one more time.

Reply:
Huh, I learned something today. Thanks for looking into it!